docs/adr/0006-filtering-system-and-parameters.md

```diff
@@ -0,0 +1,253 @@
+
+```php
+use ApiPlatform\Metadata\GetCollection;
+use ApiPlatform\Doctrine\Common\AndParameter;
+
+#[GetCollection(parameters: ['and' => new AndParameter])]
+#[AndParameter('and')]
+class Book {}
+```
+
+API Platform will continue to provide parsed query parameters and set an `_api_query_parameters` Request attribute, in the end the filter may or may not use it:
+
+```php
+$queryString = RequestParser::getQueryString($request);
+$request->attributes->set('_api_query_parameters', $queryString ? RequestParser::parseRequestParams($queryString) : []);
+```
+
+On top of that we will provide an additional `_api_header_parameters` as we would like to introduce a `QueryParameter` and an `HeaderParameter`. 
+
+### Parameter Provider
+
+During the `Provider` phase (`RequestEvent::REQUEST`), we could use a `ParameterProvider`:
+
+```php
+/**
+ * Optionnaly transforms request parameters and provides modification to the current Operation.
+ *
+ * @implements ProviderInterface<HttpOperation>
+ */
+interface ParameterProvider extends ProviderInterface {
+    public function provider(Operation $operation, array $uriVariables = [], array $context = []): HttpOperation;
+}
+```
+
+This provider can: 
+
+1. alter the HTTP Operation to provide additional context:
+
+```php
+class GroupsParameterProvider implements ProviderInterface {
+    public function provider(Operation $operation, array $uriVariables = [], array $context = []): HttpOperation 
+    {
+        $request = $context['request'];
+        return $operation->withNormalizationContext(['groups' => $request->query->all('groups')]);
+    }
+}
+```
+
+2. alter the parameter context:
+
+```php
+class UuidParameter implements ProviderInterface {
+    public function provider(Operation $operation, array $uriVariables = [], array $context = []): HttpOperation 
+    {
+        $request = $context['request'];
+        $parameters = $request->attributes->get('_api_query_parameters');
+        foreach ($parameters as $key => $value) {
+            $parameter = $operation->getParameter($key);
+            if (!$parameter) {
+                continue;
+            }
+
+            if (!in_array('uuid', $parameter->getSchema()['type'])) {
+                continue;
+            }
+
+            // TODO: should handle array values
+            try {
+                $parameters[$key] = Uuid::fromString($value);
+            } catch (\Exception $e) {}
+            
+            if ($parameter->getFilter() === SearchFilter::class) {
+                // Additionnaly, we are now sure we want an uuid filter so we could change it:
+                $operation->withParameter($key, $parameter->withFilter(UuidFilter::class));
+            }
+        }
+
+        return $operation;
+    }
+}
+```
+
+3. Validate parameters through the ParameterValidator.
+
+### Filters
+
+Filters should remain mostly unchanged, the current informations about the `property` to filter should be specified inside a `Parameter`'s `context`.
```

Comment:
To me properties mapping should remain a first class citizen because of Hydra and because it's the most common use case.